aw/ViewControllers/Reaches/Detail/ReachDetailViewController.swift

```diff
@@ -0,0 +1,225 @@
+    }
+
+    @IBAction func learnMoreTapped(_ sender: Any) {
+        guard let reach = reach, let url = reach.url else { return }
+        UIApplication.shared.open(url, options: [:], completionHandler: nil)
+    }
+
+    @IBAction func shareButtonTapped(_ sender: Any) {
+        share(sender)
+    }
+}
+
+extension ReachDetailViewController {
+    func initialize() {
+        styleLabels()
+        drawInfo()
+        setupRefreshControl()
+    }
+
+    func styleLabels() {
+        nameLabel.apply(style: .Headline1)
+        sectionLabel.apply(style: .Text1)
+        updatedLabel.apply(style: .Text2)
+        flowLabel.apply(style: .Number2)
+        unitsLabel.apply(style: .Text1)
+        reccomendationLabel.apply(style: .Text1)
+
+        descriptionSectionLabel.apply(style: .Headline1)
+        descriptionLabel.apply(style: .Text1)
+        readMoreButton.titleLabel?.apply(style: .Label1)
+
+        descriptionLabel.enabledTypes = [.url]
+        descriptionLabel.handleURLTap { url in
+            print("Url tapped")
+            UIApplication.shared.open(url, options: [:], completionHandler: nil)
+        }
+
+        difficultyLabel.apply(style: .Headline1)
+        lengthLabel.apply(style: .Headline1)
+        gradientLabel.apply(style: .Headline1)
+        for label in statsLabels {
+            label.apply(style: .Text1)
+        }
+
+        for label in buttonLabels {
+            label.apply(style: .Text3)
+        }
+    }
+
+    func drawInfo() {
+        guard let reach = reach else { return }
+        nameLabel.text = reach.name
+        sectionLabel.text = reach.sectionCleanedHTML
+        descriptionSectionLabel.text = reach.sectionCleanedHTML
+        updatedLabel.text = reach.updatedString ?? "Updating Run Details"
+
+        if let lastReading = reach.lastGageReading,
+            let reading = Float(lastReading),
+            let unit = reach.unit {
+            flowLabel.text = String(format: reading == floor(reading) ? "%.0f" : "%.2f", reading)
+            flowLabel.textColor = reach.color
+            unitsLabel.text = unit
+        } else {
+            flowLabel.text = " "
+            unitsLabel.textColor = reach.color
+            unitsLabel.text = "Unknown runnability"
+        }
+
+        if reach.detailUpdated != nil {
+            if let description = reach.longDescription {
+                descriptionLabel.attributedText = description.htmlToAttributedString
+            } else {
+                descriptionLabel.text = "No description"
+            }
+
+            readMoreButton.isHidden = !descriptionLabel.isTruncated
+        } else {
+            descriptionLabel.text = "Loading"
+        }
+
+        if let photoUrl = reach.photoUrl {
+            imageView.loadFromUrlAsync(urlString: photoUrl)
+        } else {
+            imageViewHeightContraint.constant = 0
+        }
+
+
+        if let difficulty = reach.difficulty {
+            difficultyLabel.text = "Class \(difficulty)"
```

Comment:
Fyi, best practice would pull all of the user facing strings out into a localized strings file, so that the strings can be updated for different regions and languages, and easily changed independent of the code. I'll leave it up to you whether you want to do that or not.

Reply:
Does AW have any of their site localized?

Reply:
I actually have no idea.

Reply:
It doesn't look like it on the homepage, and a check of `index.html.fr` type localizing isn't showing anything.